File: src/server/src/velvet_dawn/mechanics/selectors/selector_tiles.py
from typing import Union, List

import velvet_dawn
from .selector import Selector
from ...db.instances import Instance, TileInstance, UnitInstance, WorldInstance

""" Selector 'tile' references the tile the current unit is in

Some examples:
 - {"targets": "tile.movement.weight", "add": 5}
"""


class SelectorTile(Selector):
    """ Get the tile of the position of the given instance """

    def __init__(self):
        Selector.__init__(self, selector_name="tile")

    def new(self):
        return SelectorTile()

    def get_selection(self, instance: Instance) -> List[Instance]:
        if isinstance(instance, TileInstance):
            return self.filters.filter(instance, [instance])

        if isinstance(instance, UnitInstance):
            tile = velvet_dawn.db.tiles.get_tile(x=instance.x, y=instance.y)
            if tile:
                return self.filters.filter(instance, [tile])

        return []


class SelectorTiles(Selector):
    """ Get a list of tiles """

    def __init__(self):
        Selector.__init__(self, selector_name="tiles")

    def new(self):
        return SelectorTiles()

    def get_selection(self, instance: Union[TileInstance, UnitInstance, WorldInstance]) -> List[Instance]:
        return self.filters.filter(instance, velvet_dawn.db.tiles.all())

File: src/jserver/src/velvetdawn/core/mechanics/selectors/tiles/SelectorTiles.java
package velvetdawn.core.mechanics.selectors.tiles;

import velvetdawn.core.VelvetDawn;
import velvetdawn.core.mechanics.selectors.Selector;
import velvetdawn.core.models.instances.Instance;
import java.util.Collection;

public class SelectorTiles extends Selector {

    /* Get a list of tiles */

    public SelectorTiles(VelvetDawn velvetDawn) {
        super(velvetDawn);
    }

    @Override
    protected Collection<Instance> getSelection(Instance instance) {
        return this.filters.filter(instance, velvetDawn.map.listTiles());
    }
}
